### PR TITLE
[Op][string] Update arithmetic ops type validation to disallow string type

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 - Update arithmetic ops type validation to disallow string type

### Tickets:
 - 134630

### CODE REVIEW

File: src/core/src/op/util/unary_elementwise_arithmetic.cpp [hunk 1]
```diff
@@ -12,11 +12,11 @@ ov::op::util::UnaryElementwiseArithmetic::UnaryElementwiseArithmetic(const Outpu
...
-    NODE_VALIDATION_CHECK(this,
-                          element_type.is_dynamic() || element_type != element::boolean,
-                          "Arguments cannot have boolean element type (argument element type: ",
-                          element_type,
-                          ").");
+    NODE_VALIDATION_CHECK(
+        this,
+        element_type.is_dynamic() || (element_type != element::boolean && element_type != element::string),
+        "This operation does not support input with element type: ",
+        element_type);
```
Comment (praasz) [inline src/core/src/op/util/unary_elementwise_arithmetic.cpp:19]:
```suggestion
    const auto is_supported_et = (element_type != element::boolean && element_type != element::string);
    NODE_VALIDATION_CHECK(
        this,
        element_type.is_dynamic() || is_supported_et,
        "This operation does not support input with element type: ",
        element_type);
```
To reduce condition string. The not supported precision will be in exception message.
Same for binary elementwise.

**Review by praasz — CHANGES_REQUESTED** [review]

File: src/core/src/op/util/unary_elementwise_arithmetic.cpp (hunk 1 above)
Comment (mitruska) [reply src/core/src/op/util/unary_elementwise_arithmetic.cpp]: Ok done.

File: src/core/tests/type_prop/unary_elementwise.cpp [hunk 2]
```diff
@@ -2,21 +2,25 @@
...
-        // Should have thrown, so fail if it didn't
-        FAIL() << "Did not detect incorrect element types for arithmetic operator";
-    } catch (const NodeValidationFailure& error) {
-        EXPECT_HAS_SUBSTRING(error.what(), std::string("Arguments cannot have boolean element type"));
-    } catch (...) {
-        FAIL() << "Deduced type check failed for unexpected reason";
+TEST(type_prop, unary_arithmetic_unsupported_input_element_types) {
+    {
+        auto tv0_2_4_param = make_shared<ov::op::v0::Parameter>(element::boolean, Shape{2, 4});
+        OV_EXPECT_THROW(auto bc = make_shared<op::v0::Negative>(tv0_2_4_param),
```
[the remark this answers is not in the document]
Comment (mitruska) [reply src/core/tests/type_prop/unary_elementwise.cpp]: Sure, done.

File: src/core/tests/type_prop/unary_elementwise.cpp [hunk 3]
```diff
@@ -2,21 +2,25 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#include "common_test_utils/test_assertions.hpp"
 #include "common_test_utils/type_prop.hpp"
 #include "openvino/op/negative.hpp"
 
 using namespace std;
 using namespace ov;
+using namespace testing;
```
Comment (praasz) [inline src/core/tests/type_prop/unary_elementwise.cpp:11]:
```suggestion
using testing::HasSubstr;
```
If required for `HasSubstr`

**Review by praasz — APPROVED** [review]

**Review by t-jankowski — APPROVED** [review]

File: src/core/tests/type_prop/unary_elementwise.cpp (hunk 3 above)
Comment (mitruska) [reply src/core/tests/type_prop/unary_elementwise.cpp]: Ok, done

**Pull request merged**